src/loaders/common/loader_helper.py

```diff
@@ -179,69 +186,138 @@ def is_upa_info_complete(upa_dir: str):
     return True
 
 
-def make_collection_source_dir(
-        root_dir: str,
-        env: str,
-        collection: str,
-        source_ver: str
-) -> str:
+def make_job_dir(root_dir, username):
+    """Helper function that creates a job_dir for a user under root directory."""
+    job_dir = os.path.join(root_dir, SDK_JOB_DIR, username, uuid.uuid4().hex)
+    os.makedirs(job_dir, exist_ok=True)
+    # only user can read, write, or execute
+    os.chmod(job_dir, stat.S_IRWXU)
+    return job_dir
+
+
+def make_job_data_dir(job_dir):
+    """Helper function that creates a temporary directory for sharing files beteween the host and container."""
+    data_dir = os.path.join(job_dir, "workdir/tmp")
```

Comment:
How this interacts with the containers is really confusing, as we've discovered, so we should document exactly how it works. LMK if I don't have this right.

```suggestion
def make_job_data_dir(job_dir):
    """
    Helper function that creates a temporary directory for sharing files between the host, callback server, and container.
    
    SDK modules (like AssemblyUtil) have the shared directory mounted in the container at `/kb/module/work`. The
    scratch directory provided to the SDK module `*Impl.py` code is `/kb/module/work/tmp`. The SDK code is expected
    to read and write shared files there.
    
    The callback server mounts `<job_dir>\workdir` as the host shared directory into the SDK module.
    
    `<job_dir>` is also mounted into the callback server and it writes job information (e.g. the token and job configuration) 
    into `<job_dir>\workdir` 
    """
    data_dir = os.path.join(job_dir, "workdir/tmp")
```

Reply:
what does `*Impl.py` stand for?

Reply:
E.g. https://github.com/kbaseapps/AssemblyUtil/blob/develop/lib/AssemblyUtil/AssemblyUtilImpl.py

Reply:
:+1:

Reply:
Updated!

Reply:
Oops, should be `<job_dir>/workdir` not `<job_dir>\workdir`

Reply:
Updated!

Reply:
:+1: